### ISSUE TITLE
Runner pods crashlooping 

### ISSUE THREAD

**Issue by Anusha-Kolli** [evryfs/github-actions-runner-operator#191]
Hello @davidkarlsen ,
[ghoperator.docx](https://github.com/evryfs/github-actions-runner-operator/files/5967881/ghoperator.docx)
[ghrunner.docx](https://github.com/evryfs/github-actions-runner-operator/files/5967882/ghrunner.docx)

We deployed gh operator and crd in our kubernetes cluster it's fine for few days. But today morning we noticed runner pods are crashing with below logs.

```
Runner update in progress, do not shutdown runner.
Downloading 2.277.1 runner
Waiting for current job finish running.
Generate and execute update script.
Runner will exit shortly for update, should back online within 10 seconds 
```

I am attaching runner logs and gh operator logs here for reference.

We are using image ```quay.io/evryfs/github-actions-runner:ubuntu20-20201210.0-2.276.1``` for runner  and ghoperator 2.5.4 version.

Please help us to debug the issue.

Thanks,
Anusha

**Comment by davidkarlsen** [comment]
Please upgrade to latest runner image and you should be fine.

**Comment by Anusha-Kolli** [comment]
@davidkarlsen why is runner trying to download new image even we used tagged image?

**Comment by davidkarlsen** [comment]
Because you run an old version. Please see the link I provided.

**Comment by fl-max** [comment]
@davidkarlsen is it not possible to version lock the images using this operator?

**Comment by davidkarlsen** [comment]
the operator merely schedules pods, and yes, they are version tagged (see links provided)

**Comment by fl-max** [comment]
@davidkarlsen I was referring to locking our runner image version. Read through the post in issue #246 you referenced and I think this is the culprit. So at this time, you cannot version lock the runners and they must be allowed to update else they will fail. Unless I'm missing something from that thread?

If I understand right, setting `runner.tag` to something other than latest will eventually cause the whole thing to fail (upon new release) **unless** you had a PVC for the runner binaries to be persisted after restart.

For example, if I set `runner.tag: ubuntu20-20201210.0-2.276.1` and `n` days later tag `ubuntu20-20201210.0-2.277.0` is released, the Pod (v2.276.1) will try to update itself to v2.2770. But because there's no PVC to persist this update, it begins looping. Setting the `runner.tag: latest` or having the proper PVCs avoids this?

Is there no way to prevent auto-update? I'd much rather a build fail than run updated code that hasn't been evaluated.

**Comment by davidkarlsen** [comment]
Unfortunately until GH has a setting to avoid auto-update it cannot be avoided. However this is in practice mitigated by providing updated images, containing the updated runner, very soon after their initial release (hours after it becomes available).
